### PR TITLE
(for 1.19) Drop support for Endpoints

### PR DESCRIPTION
**Pull request by howardjohn**
Since k8s 1.21, EndpointSlice has been GA. This is a superset of functionality, and is what all Istio users only k8s 1.21+ have already defaulted to.

This PR makes it the only mode, to cleanup the complexity of having multiple. Note that k8s itself mirrors Endpoints to EndpointSlices, so this is internal-only change. Kube-proxy already only watches endpointslice.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@howardjohn part of our customers are still using older k8s than 1.19. Can keep it one more release.

Would suggest changing detect endpoint mode only in 1.18. We can make `PILOT_USE_ENDPOINT_SLICE` = true

**Comment by howardjohn** [comment]
We dropped support for K8s 1.19 long ago: https://istio.io/latest/docs/releases/supported-releases/. This is not introducd by this PR, its already unsupported and has other issues.

Kubernetes 1.19 is 3 years old. If a user wants to use a cutting-edge Istio release (this is 1.19, so its still 4 months out as well), they should use a k8s version that is more than 3 years old.

File: pilot/pkg/serviceregistry/kube/controller/endpointslice.go [hunk 1]
```diff
@@ -112,53 +106,31 @@ func serviceNameForEndpointSlice(labels map[string]string) string {
...
-	}
-	for _, svc := range c.servicesForNamespacedName(esc.getServiceNamespacedName(ep)) {
-		pod := c.pods.getPodByProxy(proxy)
-		builder := NewEndpointBuilder(c, pod)
-
-		discoverabilityPolicy := c.exports.EndpointDiscoverabilityPolicy(svc)
-
-		for _, port := range ep.Ports {
-			if port.Name == nil || port.Port == nil {
+	for _, svc := range esc.c.servicesForNamespacedName(getServiceNamespacedName(ep)) {
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/serviceregistry/kube/controller/endpointslice.go:111]: No but its already handled in `endpointCache` so I don't think we need it here either

**Comment by Purushotham233** [comment]
@howardjohn  This PR had release-notes-non label. Thus not listed in 1.19 release notes. Any reason?

JFYI, We observed a difference b/w k8s Endpoints API and EndpointSlice API in handling the terminating pods. For other reasons it was important for us to be aware of this change. Release notes is something we rely on for triaging before the upgrades.

**Comment by ramaraochavali** [comment]
Please read the description of the PR which has details on why release notes was not added - Specifically this part "Note that k8s itself mirrors Endpoints to EndpointSlices, so this is internal-only change. Kube-proxy already only watches endpointslice."

BTW, what difference are you seeing between Endpoint API and EndpointSlice API? Which K8s version and which Istio version? From which version of Istio are you upgrading to which version of Istio?

**Pull request merged**